chlog-new Makefile task fails when branch name has a `.`

**Describe the bug**

The `chlog-new` task adds a new changelog note based on the branch name: https://github.com/open-telemetry/opentelemetry-collector/blob/b45d0ed7f3c7f897fa59071a36371592c5f6d6a9/Makefile#L450-L453

If the branch name has a `.`, `chloggen` will interpret this as an extension, and, if the extension is not yaml, it will fail.

**Steps to reproduce**

1. Create branch `my.branch`
2. Run `make chlog-new`

**What did you expect to see?**

The task succeeds.

**What did you see instead?**

The task fails:

```
❯ make chlog-new
cd ./internal/tools && go install go.opentelemetry.io/build-tools/chloggen
chloggen new --filename my.branch
Error: non-yaml extension: .branch
Usage:
  chloggen new [flags]

Flags:
  -f, --filename string   name of the file to add
  -h, --help              help for new

Global Flags:
      --chloggen-directory string   directory containing unreleased change log entries (default: .chloggen)

Error: non-yaml extension: .branch
make: *** [Makefile:453: chlog-new] Error 1

```

--- a/Makefile
+++ b/Makefile
@@ -447,7 +447,7 @@ crosslink:
 chlog-install:
 	cd $(TOOLS_MOD_DIR) && $(GOCMD) install go.opentelemetry.io/build-tools/chloggen
 
-FILENAME?=$(shell git branch --show-current)
+FILENAME?=$(shell git branch --show-current).yaml
 .PHONY: chlog-new
 chlog-new: chlog-install
 	chloggen new --filename $(FILENAME)

